### PR TITLE
For Duplex scan we need "--adf-mode" parameter instead of changing source

### PR DESCRIPTION
**Pull request by marrobHD**

### CODE REVIEW

File: src/scanner.py [hunk 1]
```diff
@@ -17,14 +17,15 @@ def scan(self, front=True):
             suffix = '_front' if front else '_back'
         filename = f'"{self.config.name}"_%04d{suffix}.{self.config.scan_format}'
         filepath = self.scan_folder / filename
-        source = f'"{self.config.source_duplex}"' if self.config.duplex else self.config.source
+        source = f'"{self.config.source}"'
+        adf_mode = f'Duplex' if self.config.duplex else Simplex
```
Comment (eresturo) [inline src/scanner.py:21]: Simplex is not defined, you propably meant 'Simplex'

**Review by eresturo — COMMENTED** [review]
the problem here is, that some `scanimage` parameters are depending on your scanner.

can you send me the output of your `scanimage --help`, so we can compare ?

Here is mine (HP 5590):

```
Usage: scanimage [OPTION]...

Start image acquisition on a scanner device and write image data to
standard output.

Parameters are separated by a blank from single-character options (e.g.
-d epson) and by a "=" from multi-character options (e.g. --device-name=epson).
-d, --device-name=DEVICE   use a given scanner device (e.g. hp:/dev/scanner)
    --format=pnm|tiff|png|jpeg|pdf  file format of output file
-i, --icc-profile=PROFILE  include this ICC profile into TIFF file
-L, --list-devices         show available scanner devices
-f, --formatted-device-list=FORMAT similar to -L, but the FORMAT of the output
                           can be specified: %d (device name), %v (vendor),
                           %m (model), %t (type), %i (index number), and
                           %n (newline)
-b, --batch[=FORMAT]       working in batch mode, FORMAT is `out%d.pnm' `out%d.tif'
                           `out%d.png' or `out%d.jpg' by default depending on --format
                           This option is incompatible with --output-file.    --batch-start=#        page number to start naming files with
    --batch-count=#        how many pages to scan in batch mode
    --batch-increment=#    increase page number in filename by #
    --batch-double         increment page number by two, same as
                           --batch-increment=2
    --batch-print          print image filenames to stdout
    --batch-prompt         ask for pressing a key before scanning a page
    --accept-md5-only      only accept authorization requests using md5
-p, --progress             print progress messages
-o, --output-file=PATH     save output to the given file instead of stdout.
                           This option is incompatible with --batch.
-n, --dont-scan            only set options, don't actually scan
-T, --test                 test backend thoroughly
-A, --all-options          list all available backend options
-h, --help                 display this help message and exit
-v, --verbose              give even more status messages
-B, --buffer-size=#        change input buffer size (in kB, default 32)
-V, --version              print version information
Output format is not set, using pnm as a default.

Options specific to device `hp5590:libusb:002:027':
    -l 0..215.889mm (in steps of 0.0999908) [0]
        Top-left x position of scan area.
    -t 0..297.699mm (in steps of 0.0999908) [0]
        Top-left y position of scan area.
    -x 0..215.889mm (in steps of 0.0999908) [215.889]
        Width of scan-area.
    -y 0..297.699mm (in steps of 0.0999908) [297.699]
        Height of scan-area.
    --mode Color|Color (48 bits)|Gray|Lineart [Lineart]
        Selects the scan mode (e.g., lineart, monochrome, or color).
    --source Flatbed|ADF|ADF Duplex|TMA Slides|TMA Negatives [Flatbed]
        Selects the scan source (such as a document-feeder).
    --resolution 100|200|300|600|1200|2400dpi [100]
        Sets the resolution of the scanned image.
    --extend-lamp-timeout[=(yes|no)] [no] [advanced]
        Extends lamp timeout (from 15 minutes to 1 hour)
    --wait-for-button[=(yes|no)] [no]
        Waits for button before scanning
    --preview[=(yes|no)] [no]
        Request a preview-quality scan.
    --hide-eop-pixel[=(yes|no)] [yes] [advanced]
        Hide end-of-page indicator pixels and overwrite with neighbor pixels
    --trailing-lines-mode raw|last|raster|white|black|color [last] [advanced]
        raw = raw scan data, last = repeat last scan line, raster = b/w
        raster, white = white color, black = black color, color = RGB or gray
        color value
    --trailing-lines-color <int> [8323199] [advanced]
        Color value for trailing lines filling mode 'color'. RGB color as
        r*65536+256*g+b or gray value (default=violet or gray)

Type ``scanimage --help -d DEVICE'' to get list of all options for DEVICE.

```

File: defaults.conf [hunk 2]
```diff
@@ -1,5 +1,4 @@
 resolution = 300
-source = ADF
-source_duplex = ADF Duplex
+source = "Automatic Document Feeder"
```
[the remark this answers is not in the document]
Comment (marrobHD) [reply defaults.conf:2]: Well except HP devices...

[the remark this answers is not in the document]
Comment (eresturo) [reply defaults.conf:2]: ok, i agree in changing the default, thanks! :slightly_smiling_face:

**Comment by eresturo** [comment]
ok, i will merge this now, since i want to support hp and epson scanners, of course :slightly_smiling_face:  

i will try to find a somehow "nice" option to let users configure both scanners (and probably others)

**Comment by marrobHD** [comment]
@eresturo Maybe we should add manufacturer specific parameters for scanimage in the script which then can be manually called via `manufacturer=epson | hp | canon [auto identify manufacturer, defaults to hp]` parameter for scan.py.

**Comment by eresturo** [comment]
sorry i missed your proposal here before implementing the "command config".

you are right, one could do somehow automatically. We could scan sane's documentation to be compatible with every scanner, but i am too lazy :laughing: 

It's a command line tool, so propably only nerds are using it... They will figure it out, i guess.

**Pull request merged**